src/coreclr/jit/importer.cpp

```diff
@@ -4387,6 +4387,9 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
     FieldSeq::FieldKind fieldKind =
         isSharedStatic ? FieldSeq::FieldKind::SharedStatic : FieldSeq::FieldKind::SimpleStatic;
 
+    bool hasKnownDirectAddress = !opts.IsReadyToRun() && pFieldInfo->fieldLookup.addr != nullptr &&
```

Comment:
This condition is not pretty. I think it would look much better to just return the address in fieldLookup for all static address cases. Something like this: https://github.com/jkotas/runtime/commit/9301a2980117f7f5ba26dd78ad334c4e20bdc497

Reply:
(feel free to cherry-pick my commit if you like it)

Reply:
Multiple places in the JIT call `getFieldAddress`. What guarantees that all these places have the same idea on whether the static is boxed or not?

When I tried to run a few tests with https://github.com/jkotas/runtime/commit/9301a2980117f7f5ba26dd78ad334c4e20bdc497 , I started hitting asserts about what looked like different parts of JIT having different idea about whether the static is boxed or not.

Reply:
@jkotas Yes, I also noticed that we call getFieldAddress right after getFieldinfo for certain cases and could merge that but I intentionally limited this to my optimization so we either import ldsfld(a)/stsfld(a) early to IND(CNS) for known (initialized) statics or leave everything as it was. Can we make a clean up separately? If you insist I can look into doing it as part of this PR

Reply:
I seems `getFieldAddress` becomes an API of questionable utility after these changes. So I would also be in favor of switching to `getFieldInfo` providing the address in all cases, and deleting `getFieldAddress` (the two its uses in emitter are dead code and be safely removed).

Reply:
I understand but it's used in 5 places (including two in the emitter) so I'd love to audit them and do "remove getFieldAddress" work separately as it's not related to this PR that just improves boxed structs.

Reply:
I am not convinced that the getFieldAddress cleanup can be done separately after this PR. (It can be done seperately before this PR.)

The problem that I am worried about is that getFieldInfo returns the unboxed address and the JIT later decides to re-fetch the getFieldAddress again, gets the boxed address and the generated code will be wrong.

Reply:
Ok, will take a look tomorrow

Reply:
I'm also a fan of getting rid of `getFieldAddress`. Last time I wanted to improve handling of static fields in NativeAOT, it was kind of in the way (https://github.com/dotnet/runtime/pull/63620#issuecomment-1014110023). This would likely help.